samples/amber.cc

```diff
@@ -325,10 +326,18 @@ int main(int argc, const char** argv) {
 
     if (!options.image_filename.empty()) {
       std::string image;
+      bool usePNG = options.image_filename.compare(
+          options.image_filename.length() - 5, options.image_filename.length(),
```

Comment:
What if the string is very short, e.g. length is 2, then 2-5 is large negative that is then converted to unsigned....  out of bounds?

Reply:
done